shaka-streamer

```diff
@@ -60,15 +60,16 @@ def main():
                            'bitrates and resolutions for transcoding. ' +
                            '(optional, see example in ' +
                            'config_files/bitrate_config.yaml)')
-  parser.add_argument('-c', '--cloud-url',
-                      default=None,
-                      help='The Google Cloud Storage or Amazon S3 URL to ' +
-                           'upload to.  (Starts with gs:// or s3://)')
   parser.add_argument('-o', '--output',
                       default='output_files',
                       help='The output folder to write files to, or an HTTP ' +
                            'or HTTPS URL where files will be PUT.' +
                            'Used even if uploading to cloud storage.')
+  parser.add_argument('-H', '--add-header',
```

Comment:
I found it difficult to figure out how to get the necessary headers for a refresh token.  I can enable oauth2 in GCP, get a client-id and client-secret, but I don't understand how to get an initial refresh-token in an easy way.

Here are some ideas:

1. Streamer could get the initial refresh token for me, just based on client-id and client-secret
2. There would be docs (maybe in the examples of docs/source/overview.rst) showing how to get refresh-token
3. We could use a library like [google-auth](https://google-auth.readthedocs.io/en/master/user-guide.html) to handle the authentication (haven't found the s3 equivalent yet)

I found a quick and easy way to make it work with this code, with gcloud already installed and authenticated, but I don't think it's a "best practice":

```
  -H "Authorization=Bearer $(gcloud auth print-access-token)" \
  -o gs://my_gcs_bucket/folder/
```

It seems like this might be equivalent to using "application default credentials" in google-auth:

```py
import google.auth
credentials, project = google.auth.default()
```

Reply:
It is indeed not user friendly and needs a rework. I would go for the option of using a client id+secret, the third option is ambiguous to me right now but I will look into it.

BTW, I was getting refresh tokens from [google's oauth 2.0 playground](https://developers.google.com/oauthplayground/).